docs/src/main/sphinx/functions/table.rst

```diff
@@ -0,0 +1,69 @@
+===============
+Table functions
+===============
+
+A table function is a function returning a table. It can be invoked inside the
+``FROM`` clause of a query::
+
+    SELECT * FROM my_function(1, 100)
+
+The row type of the returned table can depend on the arguments passed with
+invocation of the function. If different row types can be returned, the
+function is a **polymorphic table function**.
+
+Polymorphic table functions allow you to dynamically invoke custom logic from
+within the SQL query. They can be used for working with external systems as
+well as for enhancing Trino with capabilities going beyond the SQL standard.
+
+Trino supports adding custom table functions. They are declared by connectors
+through implementing dedicated interfaces.
+
+Table function invocation
+-------------------------
+
+You invoke a table function in the ``FROM`` clause. Table function invocation
+syntax is similar to a scalar function call.
+
+Function resolution
+^^^^^^^^^^^^^^^^^^^
+
+Every table function is provided by a catalog, and it belongs to a schema in
+the catalog. You can qualify the function name with a schema name, or with
+catalog and schema names::
+
+    SELECT * FROM schema_name.my_function(1, 100)
+    SELECT * FROM catalog_name.schema_name.my_function(1, 100)
+
+Otherwise, the standard Trino name resolution is applied. The connection
+between the function and the catalog must be identified, because the function
+is executed by the corresponding connector. If the function is not registered
+by the specified catalog, the query fails.
+
+The table function name is resolved case-insensitive, analogically to scalar
+function and table resolution in Trino.
+
+Argument passing conventions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+There are two conventions of passing arguments to a table function:
+
+- arguments passed by name::
```

Comment:
Maybe we should start with uppercase .. and maybe bold it .. whatever you think looks good in the output